### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by robbyoconnor**
This addresses a major issue we have. Thanks to github for this feature!

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Thank you for your pull request. In order for your pull request to be accepted please check the following.
+
+- [ ] I have Contribution Guidelines at https://github.com/syl20bnr/spacemacs/tree/develop/CONTRIBUTING.org
```
Comment (d12frosted) [inline .github/PULL_REQUEST_TEMPLATE]: I have read, maybe?

File: .github/PULL_REQUEST_TEMPLATE [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+- [ ] My pull request is against the `develop` branch and not `master`
+- [ ] I have rebased my contribution against `develop` and not `master`
```
Comment (d12frosted) [inline .github/PULL_REQUEST_TEMPLATE]: I think it might be not clear to users. Though not sure what is the best way to say this :D

**Comment by d12frosted** [comment]
A quick one, nice :)

BTW, I still not convinced that it's best solution. It's a pity that we are not allowed to configure PRs using whitelists and blacklists of branches.

File: .github/PULL_REQUEST_TEMPLATE (hunk 2 above)
Comment (robbyoconnor) [inline .github/PULL_REQUEST_TEMPLATE]: If people can't understand this, they have a mental deficiency.

Comment (robbyoconnor) [inline .github/PULL_REQUEST_TEMPLATE]: fixed.

**Comment by robbyoconnor** [comment]
@d12frosted it's literally the only way people will do things -- if it's RIGHT in front of them...they read it. If they STILL fail to read it, then I reserve the right to call them morons.

**Comment by robbyoconnor** [comment]
@d12frosted it's literally the only way people will do things -- if it's RIGHT in front of them...they read it. If they STILL fail to read it, then I reserve the right to call them morons.  They're linked the contribution guidelines, yet few people seem to freaking read them, ever. I believe we should strictly enforce our contribution guidelines.

**Comment by d12frosted** [comment]
I mean, I would like to have a system that just doesn't allow you to send PR agains other branch than develop

**Comment by robbyoconnor** [comment]
That'd be nice, but baby steps. Github is responding and addressing the pain points, so let's be grateful for at least that.

**Comment by robbyoconnor** [comment]
@syl20bnr -- expedite this PR :)

**Comment by syl20bnr** [comment]
Sure I will merge it but expect a drastic edit of it. This version is too verbose and too scary. I want people to contribute first, not learn the whole git story ;-) 

> _Before submitting your Pull Request be sure to select the develop branch._

That's all we need.

**Comment by robbyoconnor** [comment]
@syl20bnr edited! See  44e5041

**Comment by robbyoconnor** [comment]
@syl20bnr, @TheBB

**Comment by robbyoconnor** [comment]
@syl20bnr do I need to fix this?

**Comment by syl20bnr** [comment]
Thank you! :+1: 
Cherry-picked into develop branch, you can safely delete your branch.

**Pull request closed without merging**